src/fairseq2/recipes/trainer.py

```diff
@@ -580,6 +580,12 @@ def _maybe_restore_state(self) -> None:
 
         log.info("Training restored. Resuming.")
 
+        try:
+            self._unit.set_step_nr(step_nr)
+            self._unit.maybe_sync_models(force_sync=True)
+        except:
+            pass
+
```

Comment:
This probably shouldn't be done here. But I can't do it directly in the finetune unit creation, because it has to be after the `step_nr` is loaded.